src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -445,3 +452,15 @@ def _parse_flow_file(self):
                 target=ErrorTarget.EXECUTOR,
             ) from e
         return module_name, func_name
+
+    def _init_input_sign(self):
+        if not self.is_function_entry:
+            with open(self._working_dir / self._flow_file, "r", encoding="utf-8") as fin:
+                flow_dag = load_yaml(fin)
+            flow = FlexFlow.deserialize(flow_dag)
+            self._inputs_sign = flow.inputs
+            self._init_sign = flow.init
+        else:
+            # For function entry there is no yaml file to get the inputs and init signature.
+            self._inputs_sign = {}
```

Comment:
We may add comments to explain what is inputs_sign and init_sign

Reply:
Added.